File: python_mid_june_2024/13. Roman to Integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        normals = {'I': 1, 'V': 5, 'X': 10, 'L': 50,
                  'C': 100, 'D': 500, 'M': 1000}
        ans = 0
        prevalue = 0
        for c in reversed(s):
            b = normals[c]
            if b >= prevalue:
                ans += b
            else:
                ans -= b
            prevalue = b
        return ans

File: java_july/13. Roman to Integer.java
class Solution {
    public int romanToInt(String s) {
        int n = s.length();
        int res = 0;
        for (int i=0; i<n; i++) {
            switch(s.charAt(i)){
                case 'I':
                    if (i+1 < n && (s.charAt(i+1)=='V' || s.charAt(i+1)=='X')){
                        res -= 1;
                    } else {
                        res += 1;
                    }
                    break;
                case 'X':
                    if (i+1 < n && (s.charAt(i+1)=='L' || s.charAt(i+1)=='C')){
                        res -= 10;
                    } else {
                        res += 10;
                    }
                    break;
                case 'C':
                    if (i+1 < n && (s.charAt(i+1)=='D' || s.charAt(i+1)=='M')){
                        res -= 100;
                    } else {
                        res += 100;
                    }
                    break;
                case 'V':
                    res += 5;
                    break;
                case 'L':
                    res += 50;
                    break;
                case 'D':
                    res += 500;
                    break;
                case 'M':
                    res += 1000;
                    break;
            }
        }
        return res;
    }
}